### PR TITLE
fix node distribution among shards

### PR DESCRIPTION
**Pull request by robin-thomas**
## Description
* Fix issue https://github.com/Zilliqa/Zilliqa/issues/765

### CODE REVIEW

File: src/libDirectoryService/DSBlockPreProcessing.cpp [hunk 1]
```diff
@@ -201,9 +204,9 @@ void DirectoryService::ComputeSharding(const VectorOfPoWSoln& sortedPoWSolns) {
                                     << DataConversion::charArrToHexStr(kv.first)
                                     << endl);
     }
+
     const PubKey& key = kv.second;
-    auto& shard =
-        m_shards.at(min(i / m_mediator.GetShardSize(false), max_shard));
+    auto& shard = m_shards.at(i / numNodesPerShard);
```
[the remark this answers is not in the document]
Comment (robin-thomas) [reply src/libDirectoryService/DSBlockPreProcessing.cpp]: Fixed that. Can you check again?

[the remark this answers is not in the document]
Comment (KaustubhShamshery) [reply src/libDirectoryService/DSBlockPreProcessing.cpp]: Sorry for not being clear enough. Say we have 603 node, the shard size is 200, then nodes should be 201,201,201 evenly distributed as well.

[the remark this answers is not in the document]
Comment (robin-thomas) [reply src/libDirectoryService/DSBlockPreProcessing.cpp]: Oh. Got it. Thanks! Let me fix that

File: src/libDirectoryService/DSBlockPreProcessing.cpp [hunk 2]
```diff
@@ -150,13 +150,16 @@ void DirectoryService::ComputeSharding(const VectorOfPoWSoln& sortedPoWSolns) {
...
   uint32_t max_shard = numOfComms - 1;
+  uint32_t numNodesPerShard = numShardNodes;
 
   if (numOfComms == 0) {
     LOG_GENERAL(WARNING, "Cannot form even one committee "
                              << " number of Pows " << sortedPoWSolns.size()
                              << " Setting numOfcomms to be 1");
     numOfComms = 1;
     max_shard = 0;
+  } else {
```
Comment (KaustubhShamshery) [inline src/libDirectoryService/DSBlockPreProcessing.cpp]: actually, no need for else here.

**Pull request merged**